var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -16,89 +16,140 @@ class Neuron(CMakePackage):
+    conflicts("%clang", when="+gpu", msg=gpu_compiler_message)
 
     patch("patch-v782-git-cmake-avx512.patch", when="@7.8.2")
 
     def cmake_args(self):
         spec = self.spec
+        args = []
+        for variant in ["backtrace", "coreneuron", "interviews", "mpi", "python", "rx3d", "tests"]:
+            args.append(self.define_from_variant("NRN_ENABLE_" + variant.upper(), variant))
 
-        def cmake_options(spec_options):
-            value = "TRUE" if spec_options in spec else "FALSE"
-            cmake_name = spec_options[1:].upper().replace("-", "_")
-            return "-DNRN_ENABLE_" + cmake_name + ":BOOL=" + value
-
-        args = [
-            cmake_options(variant)
-            for variant in [
-                "+coreneuron",
-                "+interviews",
-                "+mpi",
-                "+python",
-                "+rx3d",
-                "+coreneuron",
-                "+tests",
-            ]
-        ]
-        args.append("-DNRN_ENABLE_BINARY_SPECIAL=ON")
+        args.append(self.define_from_variant("CORENRN_ENABLE_SHARED", "shared"))
 
-        if "~mpi" in spec and "+coreneuron" in spec:
-            args.append("-DCORENRN_ENABLE_MPI=OFF")
+        if spec.satisfies("@:8"):
+            args.append(self.define("NRN_ENABLE_BINARY_SPECIAL", "ON"))
 
         if "+python" in spec:
-            args.append("-DPYTHON_EXECUTABLE:FILEPATH=" + spec["python"].command.path)
+            args.append(self.define("PYTHON_EXECUTABLE", spec["python"].command.path))
 
-        if spec.variants["build_type"].value == "Debug":
-            args.append("-DCMAKE_C_FLAGS=-g -O0")
-            args.append("-DCMAKE_CXX_FLAGS=-g -O0")
-            args.append("-DCMAKE_BUILD_TYPE=Custom")
-
-        if "+legacy-unit" in spec:
-            args.append("-DNRN_DYNAMIC_UNITS_USE_LEGACY=ON")
+        if "+legacy-unit" in spec and spec.satisfies("@:8"):
+            args.append(self.define("NRN_DYNAMIC_UNITS_USE_LEGACY", "ON"))
 
         if "+caliper" in spec:
-            args.append("-DCORENRN_CALIPER_PROFILING=ON")
+            args.append(self.define("NRN_ENABLE_PROFILING", "ON"))
+            args.append(self.define("NRN_PROFILER", "caliper"))
+
+        if spec.satisfies("+coreneuron"):
+            options = [
+                self.define("CORENRN_ENABLE_SPLAYTREE_QUEUING", "ON"),
+                self.define("CORENRN_ENABLE_TIMEOUT", "OFF"),
+                self.define_from_variant("CORENRN_ENABLE_OPENMP", "openmp"),
+                self.define_from_variant("CORENRN_ENABLE_LEGACY_UNITS", "legacy-unit"),
+                self.define_from_variant("CORENRN_ENABLE_UNIT_TESTS", "tests"),
+            ]
+
+            nmodl_options = "codegen --force"
+            if spec.satisfies("+sympy"):
+                nmodl_options += " sympy --analytic"
+            options.append(self.define("CORENRN_NMODL_FLAGS", nmodl_options))
+
+            if spec.satisfies("+gpu"):
+                nvcc = which("nvcc")
```

Comment:
Can't we use either `spack_cc` or `self.compiler.cc` (depending on whether you need the wrapper), and avoid a call to `which`?

Reply:
those will be `nvc` and `nvcc`, right? I see that we have a dependency with cuda. So I replaced it with `spec["cuda"].prefix.bin.nvcc`.